File: Java (client)/Persistence/src/main/java/ro/mpp2024/UserDBRepository.java
package ro.mpp2024;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class UserDBRepository implements IUserRepository {

    private JdbcUtils dbUtils;
    private static final Logger logger = LogManager.getLogger(UserDBRepository.class);

    public UserDBRepository(Properties props){
        logger.info("Initializing UserDBRepository with properties: {} ",props);
        dbUtils=new JdbcUtils(props);
    }

    @Override
    public User save(User entity) {
        logger.traceEntry("saving user {} ",entity);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement statement = con.prepareStatement("insert into users(name, password) values (?, ?) ")){
            statement.setString(1, entity.getName());
            statement.setString(2, entity.getPassword());
            int result = statement.executeUpdate();
            return entity;
        }catch(SQLException e){
            logger.error(e);
            System.out.println("Error DB "+e);
        }
        return null;
    }

    @Override
    public User delete(Integer aLong) {
        return null;
    }

    @Override
    public User update(User entity) {
        return null;
    }

    @Override
    public User findOne(Integer aLong) {
        logger.traceEntry("finding user with id {} ",aLong);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement statement = con.prepareStatement("select * from users where id = ?")) {
            statement.setInt(1, aLong);
            try (var result = statement.executeQuery()) {
                if (result.next()) {
                    int id = result.getInt("id");
                    String name = result.getString("name");
                    String password = result.getString("password");
                    User user = new User(id, name, password);
                    return user;
                }
            }
        }catch (SQLException e){
            logger.error(e);
            System.out.println("Error DB "+e);
        }
        return null;
    }

    @Override
    public Iterable<User> findAll() {
        logger.traceEntry("finding all users");
        List<User> users = new ArrayList<>();
        Connection con=dbUtils.getConnection();
        try(PreparedStatement statement = con.prepareStatement("select * from users")){
            try(var result = statement.executeQuery()){
                while(result.next()){
                    int id = result.getInt("id");
                    String name = result.getString("name");
                    String password = result.getString("password");
                    User user = new User(id, name, password);
                    users.add(user);
                }
            }}
        catch (SQLException e){
            logger.error(e);
            System.out.println("Error DB "+e);
        }
        return users;
    }

}

File: C# (server)/Persistence/UserDBRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using Model;
using log4net;
using log4net.Util;

namespace Persistence
{
    public class UserDBRepository : IUserRepository
    {
        private static readonly ILog logger = LogManager.GetLogger("UsersDbRepo");

        IDictionary<String, string> props;

        public UserDBRepository(IDictionary<String, string> props)
        {
            logger.Info("Creating UserDBRepository");
            this.props = props;
        }

        public IEnumerable<User> findAll()
        {
            logger.Info("Finding all users");
            IList<User> users = new List<User>();
            IDbConnection connection = DBUtils.getConnection(props);
            using (var command = connection.CreateCommand())
            {
                command.CommandText = "select id,name,password from users";
                using (var result = command.ExecuteReader())
                {
                    while (result.Read())
                    {
                        int id = result.GetInt32(0);
                        string name = result.GetString(1);
                        string password = result.GetString(2);
                        User user = new User(id, name, password);
                        users.Add(user);
                    }
                }
            }

            logger.InfoExt(users);
            return users;
        }

        public User findOne(int id)
        {
            logger.Info("Finding user with id " + id);
            IDbConnection connection = DBUtils.getConnection(props);
            using (var command = connection.CreateCommand())
            {
                command.CommandText = "select id,name,password from users where id=@id";
                IDbDataParameter paramId = command.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                command.Parameters.Add(paramId);
                using (var result = command.ExecuteReader())
                {
                    if (result.Read())
                    {
                        int idV = result.GetInt32(0);
                        string name = result.GetString(1);
                        string password = result.GetString(2);
                        User user = new User(idV, name, password);
                        logger.InfoExt(user);
                        return user;
                    }
                }
            }

            return null;
        }

        public User save(User entity)
        {
            throw new System.NotImplementedException();
        }

        public User delete(int id)
        {
            throw new System.NotImplementedException();
        }

        public User update(User entity)
        {
            throw new System.NotImplementedException();
        }
    }
}